Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -20,28 +20,23 @@ def change_rpath(file, old_prefix, new_prefix)
 
     patchelf = DevelopmentTools.locate "patchelf"
     odie "Could not locate patchelf, please: brew install patchelf." if patchelf.nil?
+    cmd = [patchelf]
 
-    cmd_rpath = [patchelf, "--print-rpath", file]
-    old_rpath = Utils.popen_read(*cmd_rpath, err: :out).strip
+    old_rpath = file.rpath
+    new_rpath = if old_rpath
+      rpath = old_rpath
+              .split(":")
+              .map { |x| x.sub(old_prefix, new_prefix) }
+              .select { |x| x.start_with?(new_prefix, "$ORIGIN") }
 
-    # patchelf requires that the ELF file have a .dynstr section.
-    # Skip ELF files that do not have a .dynstr section.
-    return if ["cannot find section .dynstr", "strange: no string table"].include?(old_rpath)
+      lib_path = "#{new_prefix}/lib"
+      rpath << lib_path unless rpath.include? lib_path
```

Comment:
Could this use a `Set` instead?

Reply:
I agree that because order matters we need an `Array` rather than a `Set`.

Reply:
Fine with me 👍.